File: LoginDemo/LoginDemo/ViewModel/Common/PasswordBoxHelper.vb
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Interactivity

''' <summary>
''' 增加Password扩展属性
''' </summary>
Public NotInheritable Class PasswordBoxHelper
        Private Sub New()
        End Sub
        Public Shared Function GetPassword(obj As DependencyObject) As String
            Return DirectCast(obj.GetValue(PasswordProperty), String)
        End Function

        Public Shared Sub SetPassword(obj As DependencyObject, value As String)
            obj.SetValue(PasswordProperty, value)
        End Sub

        Public Shared ReadOnly PasswordProperty As DependencyProperty = DependencyProperty.RegisterAttached("Password", GetType(String), GetType(PasswordBoxHelper), New PropertyMetadata("", AddressOf OnPasswordPropertyChanged))

        Private Shared Sub OnPasswordPropertyChanged(sender As DependencyObject, e As DependencyPropertyChangedEventArgs)
            Dim box As PasswordBox = TryCast(sender, PasswordBox)
            Dim password As String = DirectCast(e.NewValue, String)
            If box IsNot Nothing AndAlso box.Password <> password Then
                box.Password = password
            End If
        End Sub
    End Class

    ''' <summary>
    ''' 接收PasswordBox的密码修改事件
    ''' </summary>
    Public Class PasswordBoxBehavior
        Inherits Behavior(Of PasswordBox)
        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()

            AddHandler AssociatedObject.PasswordChanged, AddressOf OnPasswordChanged
        End Sub

        Protected Overrides Sub OnDetaching()
            MyBase.OnDetaching()

            RemoveHandler AssociatedObject.PasswordChanged, AddressOf OnPasswordChanged
        End Sub

        Private Shared Sub OnPasswordChanged(sender As Object, e As RoutedEventArgs)
            Dim box As PasswordBox = TryCast(sender, PasswordBox)
            Dim password As String = PasswordBoxHelper.GetPassword(box)
            If box IsNot Nothing AndAlso box.Password <> password Then
                PasswordBoxHelper.SetPassword(box, box.Password)
            End If
        End Sub
    End Class

File: LetMeWin/LetMeWin/Common/PasswordBoxHelper.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Interactivity;

namespace LetMeWin.Common
{
    

    /// <summary>
    /// 增加Password扩展属性
    /// </summary>
    public sealed class PasswordBoxHelper
    {
        private PasswordBoxHelper()
        {
        }
        public static string GetPassword(DependencyObject obj)
        {
            return (string)obj.GetValue(PasswordProperty);
        }

        public static void SetPassword(DependencyObject obj, string value)
        {
            obj.SetValue(PasswordProperty, value);
        }


        public static readonly DependencyProperty PasswordProperty = DependencyProperty.RegisterAttached("Password", typeof(string), typeof(PasswordBoxHelper), new PropertyMetadata("", OnPasswordPropertyChanged));
        private static void OnPasswordPropertyChanged(DependencyObject sender, DependencyPropertyChangedEventArgs e)
        {
            PasswordBox box = sender as PasswordBox;
            string password = (string)e.NewValue;
            if (box != null && box.Password != password)
            {
                box.Password = password;
            }
        }
    }

    /// <summary>
    /// 接收PasswordBox的密码修改事件
    /// </summary>
    public class PasswordBoxBehavior : Behavior<PasswordBox>
    {
        protected override void OnAttached()
        {
            base.OnAttached();

            AssociatedObject.PasswordChanged += OnPasswordChanged;
        }

        protected override void OnDetaching()
        {
            base.OnDetaching();

            AssociatedObject.PasswordChanged -= OnPasswordChanged;
        }

        private static void OnPasswordChanged(object sender, RoutedEventArgs e)
        {
            PasswordBox box = sender as PasswordBox;
            string password = PasswordBoxHelper.GetPassword(box);
            if (box != null && box.Password != password)
            {
                PasswordBoxHelper.SetPassword(box, box.Password);
            }
        }
    }

}
